src/test/javascript/spec/component/learning-paths/components/learning-path-lecture-unit.component.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { LearningPathLectureUnitComponent } from 'app/course/learning-paths/components/learning-path-lecture-unit/learning-path-lecture-unit.component';
+import { LectureUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/lectureUnit.service';
+import { VideoUnit } from 'app/entities/lecture-unit/videoUnit.model';
+import { TextUnitComponent } from 'app/overview/course-lectures/text-unit/text-unit.component';
+import { VideoUnitComponent } from 'app/overview/course-lectures/video-unit/video-unit.component';
+import { AttachmentUnitComponent } from 'app/overview/course-lectures/attachment-unit/attachment-unit.component';
+import { ExerciseUnitComponent } from 'app/overview/course-lectures/exercise-unit/exercise-unit.component';
+import { OnlineUnitComponent } from 'app/overview/course-lectures/online-unit/online-unit.component';
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { TranslateService } from '@ngx-translate/core';
+import { provideHttpClientTesting } from '@angular/common/http/testing';
+import { HttpResponse, provideHttpClient } from '@angular/common/http';
+import { of } from 'rxjs';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockSyncStorage } from '../../../helpers/mocks/service/mock-sync-storage.service';
+import { MockLocalStorageService } from '../../../helpers/mocks/service/mock-local-storage.service';
+import { MockTranslateService } from '../../../helpers/mocks/service/mock-translate.service';
+import { AlertService } from 'app/core/util/alert.service';
+import { MockAlertService } from '../../../helpers/mocks/service/mock-alert.service';
+
+describe('LearningPathLectureUnitComponent', () => {
+    let component: LearningPathLectureUnitComponent;
+    let fixture: ComponentFixture<LearningPathLectureUnitComponent>;
+    let lectureUnitService: LectureUnitService;
+    let getLectureUnitSpy: jest.SpyInstance;
+
+    const lectureUnit = new VideoUnit();
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [LearningPathLectureUnitComponent],
+            providers: [
+                provideHttpClient(),
+                provideHttpClientTesting(),
+                {
+                    provide: TranslateService,
+                    useClass: MockTranslateService,
+                },
+                {
+                    provide: LocalStorageService,
+                    useClass: MockLocalStorageService,
+                },
+                {
+                    provide: SessionStorageService,
+                    useClass: MockSyncStorage,
+                },
+                {
+                    provide: AlertService,
+                    useClass: MockAlertService,
+                },
+            ],
+        })
+            .overrideComponent(LearningPathLectureUnitComponent, {
```

Comment:
(Just a side note) I guess we would always have to overide the imports of the components for tests if we use standalone components, right?

Reply:
Overriding the components is not related to standalone components. This is just to improve testing efficiency as child components can specifically be add/removed. Thereby these components are not loaded and tested which makes the test faster in execution